lib/functions/main/check.sh

```diff
@@ -1,62 +1,77 @@
 #!/usr/bin/env bash
 
+###############################################################################
+#
+# string_is_contain()
+# Check for the presence of a word in a string if the field separator is mixed.
+# IFS can be "," comma " " space or ", " both or "\n" the end of a line.
+# usage: if string_is_contain "word1 word2,word3" "word2"; then
+#
+string_is_contain() {
+	[[ -z "$@" ]] || [[ "$#" != 2 ]] && \
+	echo "err: ${FUNCNAME[0]}: Bud argument $@" >&2 && return 1
```

Comment:
typo? - "Bud argument ..." -> should be "B**a**d argument ..." or "**Invalid** argument ..." ?

Reply:
> should be "B**a**d argument ..." or "**Invalid** argument

Invalid is correct

Reply:
```bash
string_is_contain() {
	[[ -z "$@" ]] && echo "err: ${FUNCNAME[0]}: Empty argument" >&2 && return 1
	[[ "$#" != 2 ]] && echo "err: ${FUNCNAME[0]}: Invalid argument: [$@]" >&2 && return 1
	local list=${1// /\\n}
	echo -e ${list//,/\\n} | grep -q "^$2$"
}
```
Would this option be better?

Reply:
Yes, this is better!

To be precise, in case `[[ "$#" != 2 ]]` it should say: "Invalid count of arguments ...".

If we want most self explaining function name it could be `string_arg1_contains_arg2`.

```bash
string_arg1_contains_arg2() {
	[[ -z "$@" ]] && echo "err: ${FUNCNAME[0]}: Empty argument" >&2 && return 1
	[[ "$#" != 2 ]] && echo "err: ${FUNCNAME[0]}: Invalid count of arguments: [$@]" >&2 && return 1
	local list=${1// /\\n}
	echo -e ${list//,/\\n} | grep -q "^$2$"
}
```
... I leave it to you.